app/js/directives/velocity-button.js

```diff
@@ -0,0 +1,102 @@
+ */
+angular.module("sn.velocity.snVelocityButton", [
+    "sn.velocity.snVelocity"
+])
+
+.directive("snVelocityButton",[
+    "$compile",
+    "$window",
+    /**
+     * @constructor
+     * @param   {Service} $compile   angular template compiler
+     * @param   {Service} $window    angular wrapper for window
+     */
+    function($compile, $window) {
+        return {
+            restrict: "EA",
+            scope: {
+                "onClickOn": "=?",
+                "onClickOff": "=?",
+                "onMouseEnter": "=?",
+                "onMouseLeave": "=?"
+            },
+            link: function($scope, $element){
+
+                /**
+                 * Number of times animation has run
+                 * @property {Number} animatedCount
+                 */
+                $scope.animatedCount = 0;
+
+                /**
+                 * State of click toggle
+                 * 'true' when animationCount is even
+                 * @property {Boolean} toggleState
+                 */
+                $scope.toggleState = false;
+
+                /**
+                 * Run animation on child elements
+                 * @param {Object} elementKeyframes   object of elements with keyframe arrays
+                 */
+                $scope.animate = function animate(elementKeyframes){
+
+                    angular.forEach(elementKeyframes, function(keyframes, key){
+                        var animationElement = $element[0].querySelector(key);
+                        angular.forEach(keyframes, function(value){
+                            $window.Velocity(animationElement, value.properties, value.options);
+                        });
+                    });
+
+                    $scope.animatedCount = $scope.animatedCount + 1;
+                    $scope.toggleState = !$scope.toggleState;
+
+                };
+
+                /**
+                 * Bind click animation to click event
+                 */
+                if ($scope.onClickOn || $scope.onClickOff) {
+                    $element.bind("click", function() {
+
+                        if (!$scope.toggleState) {
+                            $scope.animate($scope.onClickOn);
+                        } else if ($scope.toggleState) {
+                            $scope.animate($scope.onClickOff);
+                        }
+
+                    });
+                }
+
+                /**
+                 * Bind onMouseEnter animation to onMouseEnter event
+                 */
+                if ($scope.onMouseEnter) {
+                    $element.bind("mouseenter", function() {
+                        $scope.animate($scope.onMouseEnter);
+                    });
+                }
+
+                /**
+                 * Bind onMouseLeave animation to onMouseLeave event
+                 */
+                if ($scope.onMouseLeave) {
+                    $element.bind("mouseleave", function() {
```

Comment:
And another `bind` here. 
- You should also be clearing your listeners on `$destroy` as these events aren't strictly angular events. 
- I would also suggest always declaring you event handlers as functions saved as variables e.g. `var onEvent = function onEvent(){}` then do `$element.on('event', onEvent)`, it's makes the code neater as easy to debug as these functions won't be anonymous